src/ui/ui_settings/ui_settings_service.js

```
@@ -0,0 +1,130 @@
+    const all = await this.getAll();
+    return all[key];
+  }
+
+  async getAll() {
+    const raw = await this.getRaw();
+
+    return Object.keys(raw)
+      .reduce((all, key) => {
+        const item = raw[key];
+        const hasUserValue = 'userValue' in item;
+        all[key] = hasUserValue ? item.userValue : item.value;
+        return all;
+      }, {});
+  }
+
+  async getRaw() {
+    const userProvided = await this.getUserProvided();
+    return defaultsDeep(userProvided, this.getDefaults());
+  }
+
+  async getUserProvided(options) {
+    return hydrateUserSettings(await this._read(options));
+  }
+
+  async setMany(changes) {
+    await this._write(changes);
+  }
+
+  async set(key, value) {
+    await this.setMany({ [key]: value });
+  }
+
+  async remove(key) {
+    await this.set(key, null);
+  }
+
+  async removeMany(keys) {
+    const changes = {};
+    keys.forEach(key => {
+      changes[key] = null;
+    });
+    await this.setMany(changes);
+  }
+
+  async _write(changes) {
+    await this._callCluster('update', {
+      index: this._index,
+      type: this._type,
+      id: this._id,
+      body: {
+        doc: changes
+      }
+    });
+  }
+
+  async _read(options = {}) {
+    const interceptValue = await this._readInterceptor(options);
+    if (interceptValue != null) {
+      return interceptValue;
+    }
+
+    const {
+      ignore401Errors = false
+    } = options;
+
+    const isIgnorableError = error => (
+      error instanceof esErrors[404] ||
+      error instanceof esErrors[403] ||
+      error instanceof esErrors.NoConnections ||
+      (ignore401Errors && error instanceof esErrors[401])
+    );
+
+    try {
+      const clientParams = {
+        index: this._index,
+        type: this._type,
+        id: this._id,
+      };
+
+      const callOptions = {
+        wrap401Errors: !ignore401Errors
+      };
+
+      const resp = await this._callCluster('get', clientParams, callOptions);
+      return resp._source;
```

yep, the original code was using a single handler for the "ignored error" and "successful response" flow, now we branch on that condition and this is only for successful responses.